### PR TITLE
Correct usage of `xtol` in `ghi_from_poa_driesse_2023`

### LINKED ISSUE #1970: [BUG]: unused `xtol` argument in `ghi_from_poa_driesse_2023`
**Describe the bug**
In function [`pvlib.irradiance.ghi_from_poa_driesse_2023`](https://github.com/pvlib/pvlib-python/blob/26acf8ae894fb54ad9ef6836757ef4406d4bffae/pvlib/irradiance.py#L1529) the argument `xtol` is unused.
However, in [the docstring](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.ghi_from_poa_driesse_2023.html#pvlib.irradiance.ghi_from_poa_driesse_2023) it is specified.

Seems like a typo.

**To Reproduce**
N/A

**Expected behavior**
To be used.

**Screenshots**
N/A

**Versions:**
 - ``pvlib.__version__``: latest
 - ``pandas.__version__``: N/A
 - python: N/A

**Additional context**
I'll open a PR in short.

### PR DESCRIPTION
**Pull request by echedey-ls**
- [x] Closes #1970 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

I believe linked issue gives enough context.

### CODE REVIEW

File: pvlib/irradiance.py [hunk 1]
```diff
@@ -1513,11 +1513,14 @@ def poa_error(ghi):
...
                         disp=False,
                         )
-    except ValueError:
-        # this occurs when poa_error has the same sign at both end points
-        ghi = np.nan
-        conv = False
-        niter = -1
+    except ValueError as e:
+        # this may occur because poa_error has the same sign at both end points
+        if "f(a) and f(b) must have different signs" in e.args:
```
Comment (cwhanse) [inline pvlib/irradiance.py:1518]: Minor concern - what if scipy changes the text of the error message? I think I prefer to simply `raise e` and exit, and change the "this occurs" comment to "occurs when .... both end points, for example"

Comment (echedey-ls) [reply pvlib/irradiance.py:1518]: From reading the code I have understood that `f(a) and f(b) must have different signs` ValueErrors do need to be skipped. It probably was that @adriesse coded it to take into account some cases where the set of arguments passed onto bisect didn't make sense for it, but were generated from a reasonable input. (@adriesse can you confirm my suspicion?)

`scipy.optimize.bisect` also raises `ValueError` for other reasons, like a negative `xtol`. `ghi_from_poa_adriesse_2023` does not allow other problematic combinations in `bisect` due to the limited number of arguments you can pass onto it.

I'm testing the `xtol` argument propagates correctly by passing a nonsensical value (negative). In that case, a `ValueError` exception should be risen. This matching allows for raising that kind of exception from scipy. The opposite match can also be used (`if "xtol too small " in e.args: ...`).

I think letting the exception be risen from the scipy side hints any user to read the bisect documentation. That documentation is better in terms of what `xtol` is.

> what if scipy changes the text of the error message? 

I can change the condition to be `"different signs" in e.args` to preserve behaviour and be laxer or do the `xtol` validation internally in PVLIB. In any case, I doubt a numerical method like bisect will change the cause message in a failure like that.

**Review by cwhanse — APPROVED** [review]

File: pvlib/irradiance.py [hunk 2]
```diff
@@ -1557,8 +1561,8 @@ def ghi_from_poa_driesse_2023(surface_tilt, surface_azimuth,
     albedo : numeric, default 0.25
         Ground surface albedo. [unitless]
     xtol : numeric, default 0.01
-        Convergence criterion.  The estimated GHI will be within xtol of the
-        true value. [W/m^2]
+        Convergence criterion. The estimated GHI will be within xtol of the
+        true value. Must be positive. [W/m^2]
```
Comment (adriesse) [inline pvlib/irradiance.py:1565]: Interestingly, the bisect docstring says nonnegative, which is inconsistent with its code.

File: pvlib/irradiance.py [hunk 3]
```diff
@@ -1599,7 +1603,7 @@ def ghi_from_poa_driesse_2023(surface_tilt, surface_azimuth,
                                           solar_zenith, solar_azimuth,
                                           poa_global,
                                           dni_extra, airmass, albedo,
-                                          xtol=0.01)
+                                          xtol=xtol)
```
Comment (adriesse) [inline pvlib/irradiance.py:1606]: Good catch!

File: pvlib/irradiance.py (hunk 1 above)
Comment (adriesse) [reply pvlib/irradiance.py:1517]: > do the xtol validation internally in PVLIB.

I think this is a much better solution.

**Review by adriesse — CHANGES_REQUESTED** [review]

**Review by kandersolar — APPROVED** [review]
Thanks @echedey-ls!

**Pull request merged**